### ISSUE TITLE
Search component missing mouse action

### ISSUE THREAD

**Issue by kyjak** [carbon-design-system/carbon#6301]
![image](https://user-images.githubusercontent.com/15310250/84898124-0baaee80-b0a7-11ea-8a63-a08f7b76d5ed.png)
The search component is missing an optional button for search submit.
I suggest a validation function as optional, returning true if the text is valid for search.
If valid, an icon shows up/enters enabled state that is clickable and fires the search callback.
Bonus - Search callback should be also fired by hitting enter in the search field, if text is valid.
I think this is a common functionality and should not be implemented in carbon, as it is used on IBM sites and the look and feel should be common for all sites.
The clickable icon could be the spyglass which is now on the left, moved to the right.

**Comment by emyarod** [comment]
there has been some previous discussion about this functionality, here is a thread for reference https://github.com/carbon-design-system/carbon/issues/3310

**Comment by kyjak** [comment]
Ok, So after reading the 3310, we should probably code our own search bar or?

**Comment by emyarod** [comment]
yeah it isn't a supported pattern for the time being but the proposal may be reconsidered cc @carbon-design-system/design

**Comment by kyjak** [comment]
ok, I will discuss this on planning for ibm docs and be back with the result, but the chance is very high IBM DOCS will need this.

**Comment by kyjak** [comment]
After our planning call today, yes, confirmed, we need this change. Let me know if this comes to the roadmap or we code it on our own.

**Comment by andreancardona** [comment]
@jeanservaas @aagonzales @laurenmrice 

Hey designers! 

Per a follow up with Anna - in proposal reviews we said we would add a variation with this and it looks like we need designs for that. 

Let me know if can get designs for this!

Thank you!

**Comment by andreancardona** [comment]
Hey designers - just wanted to follow up on this :)

**Comment by laurenmrice** [comment]
Design still needs to make a design spec for this. We can accept this proposal and add this to our list of open enhancement work.

**Comment by Kyjaque** [comment]
And this took only 3 years😂
God bless IBM

**Comment by kyjak** [comment]
And this took only 3 tears 😂
God bless IBM

**Comment by mjabbink** [comment]
Holy Toledo.
